src/main/java/com/webdev/cosmo/cosmobackend/config/SecurityConfig.java

```diff
@@ -1,40 +1,58 @@
 package com.webdev.cosmo.cosmobackend.config;
 
+import com.webdev.cosmo.cosmobackend.config.properties.EndpointConfig;
+import com.webdev.cosmo.cosmobackend.security.CustomAuthenticationFilter;
+import com.webdev.cosmo.cosmobackend.security.CustomAuthenticationManager;
+import com.webdev.cosmo.cosmobackend.security.CustomAuthenticationProvider;
+import com.webdev.cosmo.cosmobackend.service.external.FacebookClient;
 import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.config.annotation.web.configurers.LogoutConfigurer;
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 import org.springframework.web.filter.OncePerRequestFilter;
 
-import java.util.Set;
-
-import static org.springframework.security.config.Customizer.withDefaults;
 
 @Configuration
 @RequiredArgsConstructor
 public class SecurityConfig {
 
-    private final OncePerRequestFilter customAuthenticationFilter;
+    private final FacebookClient facebookClient;
+
+    private final EndpointConfig endpointConfig;
+
+    @Bean
+    CustomAuthenticationProvider authenticationProvider(FacebookClient facebookClient) {
+        return new CustomAuthenticationProvider(facebookClient);
+    }
+
+    @Bean
+    OncePerRequestFilter customAuthenticationFilter() {
+        return new CustomAuthenticationFilter(endpointConfig, customAuthenticationManager());
+    }
 
+    @Bean
+    CustomAuthenticationManager customAuthenticationManager() {
```

Comment:
Use dependency injection instead of creating new class instance via method callback. In other words, simply pass AuthenticationProvider authenticationProvider as parameter (if any bean names collision will occur use @Qualifier)